Diffusion/Common/Views/PromptTextField.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  PromptTextField.swift
+//  Diffusion-macOS
+//
+//  Created by Dolmere on 22/06/2023.
+//  See LICENSE at https://github.com/huggingface/swift-coreml-diffusers/LICENSE
+//
+
+import SwiftUI
+import Combine
+import StableDiffusion
+
+struct PromptTextField: View {
+    @State private var output: String = ""
+    @State private var input: String = ""
+    @State private var typing = false
+    @State private var tokenCount: Int = 0
+    @State var isPositivePrompt: Bool = true
+    @State private var tokenizer: BPETokenizer?
+    @State private var currentModelVersion: String = ""
+
+    @Binding var textBinding: String
+    @Binding var model: String // the model version as it's stored in Settings
+
+    private let maxTokenCount = 75
+
+    private var modelInfo: ModelInfo? {
+        ModelInfo.from(modelVersion: $model.wrappedValue)
+    }
+    
+    private var filename: String? {
+        let variant = modelInfo?.bestAttention ?? .original
+        return modelInfo?.modelURL(for: variant).lastPathComponent
+    }
+    
+    private var downloadedURL: URL? {
+        if let filename = filename {
+            return PipelineLoader.models.appendingPathComponent(filename)
+        }
+        return nil
+    }
+    
+    private var packagesFilename: String? {
+        (filename as NSString?)?.deletingPathExtension
+    }
+    
+    private var compiledURL: URL? {
+        if let packagesFilename = packagesFilename {
+            return downloadedURL?.deletingLastPathComponent().appendingPathComponent(packagesFilename)
+        }
+        return nil
+    }
+    
+    
+    private var textColor: Color {
+        switch tokenCount {
+        case 0...65:
+            return .green
+        case 66...75:
+            return .orange
+        default:
+            return .red
+        }
+    }
+    
+    // macOS initializer
+    init(text: Binding<String>, isPositivePrompt: Bool, model: Binding<String>) {
+         _textBinding = text
+         self.isPositivePrompt = isPositivePrompt
+        _model = model
+    }
+    
+    //iOS initializer
+    init(text: Binding<String>, isPositivePrompt: Bool, model: String) {
+        _textBinding = text
+        self.isPositivePrompt = isPositivePrompt
+        _model = .constant(model)
+    }
+
+    var body: some View {
+        VStack {
+            HStack {
+                Spacer()
+                if !textBinding.isEmpty {
+                    Text("\(tokenCount)")
+                        .foregroundColor(textColor)
+                    Text(" / \(maxTokenCount)")
+                }
+            }
+            .onReceive(Just(textBinding)) { text in
+                updateTokenCount(newText: text)
+            }
+            .font(.caption)
```

Comment:
What do you think about putting this _below_ the textfield instead of above?

Reply:
Done in an upcoming additional commit. Also for iOS I aligned it left instead of right because the generate button is on the right and the token counter looked kind of misaligned towards the center of the screen.